### PR TITLE
Create beamtalk_string_ops runtime helper module (BT-361)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Creates `beamtalk_string_ops.erl` — the runtime helper module for String primitive methods that require more than simple Erlang BIF calls. These functions are called from compiled stdlib String module via `@primitive` codegen.

**Linear issue:** https://linear.app/beamtalk/issue/BT-361

## What was implemented

13 string helper functions, all grapheme-aware and UTF-8 correct:

| Function | Description |
|----------|-------------|
| `at/2` | 1-based grapheme access with structured error on bounds |
| `capitalize/1` | Capitalize first grapheme |
| `reverse/1` | Grapheme-aware reverse |
| `includes/2` | Substring check |
| `starts_with/2` | Prefix check |
| `ends_with/2` | Suffix check |
| `index_of/2` | Find substring position (1-based grapheme index or nil) |
| `split_on/2` | Global pattern split |
| `repeat/2` | Repeat N times |
| `as_list/1` | Convert to grapheme list |
| `each/2` | Iterate graphemes with block |
| `collect/2` | Map over graphemes with block |
| `select/2` | Filter graphemes by block predicate |

## Key changes

- **New:** `runtime/apps/beamtalk_runtime/src/beamtalk_string_ops.erl` — 13 exported functions
- **New:** `runtime/apps/beamtalk_runtime/test/beamtalk_string_ops_tests.erl` — 53 unit tests (UTF-8, emoji, edge cases)
- **New:** `tests/e2e/cases/string_ops.bt` — 16 E2E tests

## Notes

- `collect:` and `select:` on strings are intercepted by codegen intrinsics (`try_generate_list_message`) that generate `lists:map/filter` instead of calling the runtime helpers. Filed as BT-416.
- Uses structured `beamtalk_error` records for `at:` bounds errors (not bare tuples).

### CODE REVIEW

File: runtime/apps/beamtalk_runtime/src/beamtalk_string_ops.erl [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+%% @doc Capitalize the first grapheme, keep rest unchanged.
+-spec capitalize(binary()) -> binary().
+capitalize(<<>>) -> <<>>;
+capitalize(Str) when is_binary(Str) ->
+    case string:next_grapheme(Str) of
+        [First | Rest] ->
+            Upper = string:uppercase(unicode:characters_to_binary([First])),
+            RestBin = unicode:characters_to_binary(Rest),
+            <<Upper/binary, RestBin/binary>>;
+        [] -> <<>>
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_runtime/src/beamtalk_string_ops.erl]: Fixed. Wrapped `string:uppercase` result with `unicode:characters_to_binary` to ensure we always get a binary for the `<<Upper/binary, ...>>` concatenation.

File: runtime/apps/beamtalk_runtime/test/beamtalk_string_ops_tests.erl [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+at_out_of_bounds_test() ->
+    ?assertError(_, beamtalk_string_ops:at(<<"hi">>, 10)).
+
+at_zero_index_test() ->
+    ?assertError(_, beamtalk_string_ops:at(<<"hi">>, 0)).
+
+at_negative_index_test() ->
+    ?assertError(_, beamtalk_string_ops:at(<<"hi">>, -1)).
+
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_runtime/test/beamtalk_string_ops_tests.erl]: Fixed. Error assertions now verify `#beamtalk_error{kind = index_out_of_bounds, class = 'String', selector = 'at:'}` instead of wildcard `_`.
